lib/net/ssh/transport/ctr.rb

```diff
@@ -50,11 +50,16 @@ def update(data)
 
           encrypted = ""
 
-          while @remaining.bytesize >= block_size
-            encrypted += xor!(@remaining.slice!(0, block_size),
+          offset = 0
+          while (@remaining.bytesize - offset) >= block_size
+            encrypted += xor!(@remaining.slice(offset, block_size),
                               _update(@counter))
             increment_counter!
+            offset += block_size
           end
+          # only modify remaining after loop, we do not need it modified while
+          # looping
+          @remaining = @remaining.slice(offset..-1)
```

Comment:
I'd prefer `slice!` here unless this version is there is a measureable perf improvement. Also I don't think the comment above is neccesary the code itself is readable.

Reply:
yep, you're right i've overlooked that `slice!` has different return than `slice`